lib/check-tags.js

```diff
@@ -0,0 +1,41 @@
+var log = require('./out')({});
+
+function checkTags(app, mod, name) {
+  if (mod.tags && mod.tags !== '' && typeof mod.tags === 'string') {
+    mod.tags = mod.tags.split(',');
+  }
+
+  function checkMatch (A, B) {
+    return A.filter(function(tag) {
+      return (B.indexOf(tag) !== -1);
+    });
+  }
+
+  function checkMatches (A, B) {
+    var result = checkMatch(A, B);
+    if (result.length !== 0) {
+      return result;
+    } else {
+      return false;
+    }
+  }
+
+  if ((app.withTags && mod.tags && !checkMatches(mod.tags, app.withTags)) ||
+      (app.withTags && !mod.tags)) {
+    log.info(name, 'no matching tags found');
+    return true;
+  } else if (app.excludeTags && mod.tags && checkMatches(mod.tags,
+        app.excludeTags)) {
+    log.info(name, 'skipped because of tag: ' + checkMatches(mod.tags,
```

Comment:
You're calling checkMatches twice, needlessly. Why not just use a variable (as before)?